core/src/main/java/apoc/algo/PathFinding.java

```diff
@@ -19,12 +20,78 @@
 
 public class PathFinding {
 
+    public static class GeoEstimateEvaluatorPointCustom implements EstimateEvaluator<Double> {
+
+        // -- from org.neo4j.graphalgo.impl.util.GeoEstimateEvaluator
+        private static final double EARTH_RADIUS = 6371 * 1000; // Meters
+        private Node cachedGoal;
+        private final String pointPropertyKey;
+        private double[] cachedGoalCoordinates;
+        
+        public GeoEstimateEvaluatorPointCustom(String pointPropertyKey) {
+            this.pointPropertyKey = pointPropertyKey;
+        }
+        
+        @Override
+        public Double getCost( Node node, Node goal) {
+            double[] nodeCoordinates = getCoordinates(node);
+            if ( cachedGoal == null || !cachedGoal.equals( goal ) )
+            {
+                cachedGoalCoordinates = getCoordinates(goal);
+                cachedGoal = goal;
+            }
+            return distance(nodeCoordinates[0], nodeCoordinates[1],
+                    cachedGoalCoordinates[0], cachedGoalCoordinates[1] );
+        }
+        
+        private static double distance( double latitude1, double longitude1,
+                                        double latitude2, double longitude2 ) {
+            latitude1 = Math.toRadians( latitude1 );
+            longitude1 = Math.toRadians( longitude1 );
+            latitude2 = Math.toRadians( latitude2 );
+            longitude2 = Math.toRadians( longitude2 );
+            double cLa1 = Math.cos( latitude1 );
+            double xA = EARTH_RADIUS * cLa1 * Math.cos( longitude1 );
+            double yA = EARTH_RADIUS * cLa1 * Math.sin( longitude1 );
+            double zA = EARTH_RADIUS * Math.sin( latitude1 );
+            double cLa2 = Math.cos( latitude2 );
+            double xB = EARTH_RADIUS * cLa2 * Math.cos( longitude2 );
+            double yB = EARTH_RADIUS * cLa2 * Math.sin( longitude2 );
+            double zB = EARTH_RADIUS * Math.sin( latitude2 );
+            return Math.sqrt( ( xA - xB ) * ( xA - xB ) + ( yA - yB )
+                    * ( yA - yB ) + ( zA - zB ) * ( zA - zB ) );
+        }
+        // -- end from org.neo4j.graphalgo.impl.util.GeoEstimateEvaluator
+
+        private double[] getCoordinates(Node node) {
+            return ((PointValue) node.getProperty(pointPropertyKey)).coordinate();
+        }
+    }
+
     @Context
     public GraphDatabaseService db;
 
     @Context
     public Transaction tx;
 
+    @Procedure
+    @Description("apoc.algo.aStarWithPoint(startNode, endNode, 'relTypesAndDirs', 'distance','pointProp') - " +
```

Comment:
We have started to be stricter about what will be added into APOC core. Would it be possible to add this into APOC full instead?

Reply:
Yes sure, I just moved it in full